Link libstdc++ statically by default since it's very likely incompatible to
the system's version.

Explanation of the numbers:
 -1 means we should not link in libstdc++
 0  means we should link in libstdc++ if it is needed
 1  means libstdc++ is needed and should be linked in.
 2  means libstdc++ is needed and should be linked statically.

--- a/gcc/cp/g++spec.c
+++ b/gcc/cp/g++spec.c
@@ -157,7 +157,7 @@
 	    args[i] |= WITHLIBC;
 	  else
 	    /* Unrecognized libraries (e.g. -lfoo) may require libstdc++.  */
-	    library = (library == 0) ? 1 : library;
+	    library = (library == 0) ? 2 : library;
 	  break;
 
 	case OPT_pg:
@@ -171,7 +171,7 @@
 		  || strcmp (arg, "c++-cpp-output") == 0
 		  || strcmp (arg, "objective-c++") == 0
 		  || strcmp (arg, "objective-c++-cpp-output") == 0))
-	    library = 1;
+	    library = 2;
 		
 	  saw_speclang = 1;
 	  break;
@@ -181,7 +181,7 @@
 	  /* Arguments that go directly to the linker might be .o files,
 	     or something, and so might cause libstdc++ to be needed.  */
 	  if (library == 0)
-	    library = 1;
+	    library = 2;
 	  break;
 
 	case OPT_c:
@@ -249,7 +249,7 @@
 		    && (len <= 4 || strcmp (arg + (len - 4), ".HPP") != 0)
 		    && (len <= 4 || strcmp (arg + (len - 4), ".tcc") != 0)
 		    && (len <= 3 || strcmp (arg + (len - 3), ".hh") != 0))
-		  library = 1;
+		  library = 2;
 	      }
 	  }
 	  break;
